pkg/services/provisioning/dashboards/dashboard.go

```diff
@@ -33,6 +34,10 @@ type Provisioner struct {
 	provisioner        dashboards.DashboardProvisioningService
 }
 
+func (provider *Provisioner) HasDashboardSources() bool {
+	return len(provider.fileReaders) > 0
```

Comment:
is it possible that we delete dashboards from the DB if their "corresponding file" on the disk is missing?

Reply:
Yes, there is method called [CleanUpOrphanedDashboards](https://github.com/grafana/grafana/blob/main/pkg/services/provisioning/dashboards/dashboard.go#L80) that does this on every dashboard provision routine, but here we are not covering it – so will have some stale data in index for a while.

Reply:
okk, I was thinking about the `HasDashboardSources` name, specifically whether it means "database state will change once provisioning is complete" but I guess it's not so simple!

Reply:
Yep, require one more iteration in the future and only solving part of issues with provisioning. "database state will change once provisioning is complete" - I also thought about having this, but have not found an elegant way.